src/shell/data-staging.cpp

```diff
@@ -23,97 +23,116 @@ extern "C" {
 namespace fs = boost::filesystem;
 using json = nlohmann::json;
 
-enum JobspecState { invalid, valid_no_staging, valid_staging};
+enum class jobspec_state_t : int {
+    INVALID = 0,
+    VALID_NO_STAGING = 1,
+    VALID_STAGING = 2
+};
 
-static JobspecState validate_jobspec (json jobspec)
+static jobspec_state_t validate_jobspec (json &jobspec)
 {
-    if (!jobspec["attributes"].contains("user")) {
-        shell_log ("Jobspec does not contain user attributes. No staging necessary.");
-        return JobspecState::valid_no_staging;
-    }
-    if (!jobspec["attributes"]["user"].contains("data-staging")) {
-        shell_log ("Jobspec does not contain storage attributes. No staging necessary.");
-        return JobspecState::valid_no_staging;
+    json storage_attrs;
+    try {
+        storage_attrs = \
+            jobspec.at ("/attributes/system/data-staging"_json_pointer);
+    } catch (json::out_of_range& e) {
+        shell_log ("Jobspec does not contain data-staging attributes. "
+                   "No staging necessary.");
+        return jobspec_state_t::VALID_NO_STAGING;
     }
 
-    json storage_attrs = jobspec["attributes"]["user"]["data-staging"];
     std::string req_attrs[] = {"label", "granularity", "stage-in"};
     for (const json &storage_attr : storage_attrs) {
         for (const auto &req_attr: req_attrs) {
-            if (!storage_attr.contains(req_attr)) {
-                shell_log_error ("Storage entry missing %s key", req_attr.c_str());
-                return JobspecState::invalid;
+            if (!storage_attr.contains (req_attr)) {
+                shell_log_error ("Storage entry missing %s key",
+                                 req_attr.c_str());
```

Comment:
Nit: space before parenthesis.
```suggestion
                                 req_attr.c_str ());
```